### ISSUE TITLE
result accuracy difference between trt-llm and pytorch in Conv2d

### ISSUE THREAD

**Issue by beegerous** [NVIDIA/TensorRT-LLM#806]
I calculate conv2d in both pytorch and trt-llm, the result is quiet different.

```
env:
    ubuntu 20.04
    Tesla V100
    Driver Version: 470.103.01
    TensorRT-LLM:  a75618df24e97ecf92b8899ca3c229c4b8097dda
    CUDA Version: 12.2
    torch 2.1.0a0+32f93b1
    tensorrt-llm 0.6.1
    tensorrt 9.2.0.post12.dev5
```

I run the code in docker, which build from source ( https://github.com/NVIDIA/TensorRT-LLM/blob/main/docs/source/installation.md#on-systems-without-gnu-make ) 

code:

``` python
import torch
from collections import OrderedDict

import tensorrt as trt
import tensorrt_llm
from tensorrt_llm.builder import Builder
from tensorrt_llm.network import net_guard
from tensorrt_llm.runtime import Session
from tensorrt_llm.layers import Conv2d
from tensorrt_llm.module import Module

tensorrt_llm.logger.set_level("warning")

class MyConv2d(Module):
    def __init__(self, in_features, out_features, kernel_size, padding):
        super().__init__()
        self.conv = Conv2d(
            in_features, out_features,
            kernel_size=kernel_size,
            padding=padding,
            bias=True,
            dtype=trt.DataType.HALF)

    def forward(self, x):
        return self.conv(x)

if __name__ == "__main__":
    random.seed(1234)
    torch.manual_seed(1234)
    x = torch.ones([2, 1280, 3, 3], device="cuda", dtype=torch.float16)
    pyconv = torch.nn.Conv2d(
        1280, 1280,
        kernel_size=(3,3),
        padding=(1,1),
        dtype=torch.float16,
        bias=True,
        device="cuda",
    ).eval()
    y1 = pyconv(x)

    builder = Builder()
    builder_config = builder.create_builder_config(
        name="MyConv2d",
        precision="float16",
        opt_level=0,
        strongly_typed=True,
    )

    myconv = MyConv2d(1280, 1280, (3,3), (1,1))
    myconv.conv.weight.value = pyconv.weight
    myconv.conv.bias.value = pyconv.bias

    network = builder.create_network()

    with net_guard(network):
        trt_x = tensorrt_llm.Tensor(
            name='x',
            dtype=trt.float16,
            shape=[2, 1280, 3, 3],
            dim_range=OrderedDict([
                ('x0', [[2, 2, 2]]),
                ('x1', [[1280, 1280, 1280]]),
                ('x2', [[3, 3, 3]]),
                ('x3', [[3, 3, 3]]),
            ]))
        output = myconv(x=trt_x)
        output.mark_output("y", trt.float16)
        network.set_named_parameters(myconv.named_parameters())

    tensorrt_llm.graph_rewriting.optimize(network)
    engine = builder.build_engine(network, builder_config)
    assert engine is not None, 'Failed to build engine.'

    session = Session.from_serialized_engine(engine)
    inputs = { "x" : x }
    outputs = { "y" : torch.empty([2, 1280, 3, 3], dtype=torch.float16, device='cuda') }
    stream = torch.cuda.current_stream().cuda_stream
    ok = session.run(inputs, outputs, stream)
    assert ok
    y2 = outputs["y"]

    diff = (y1 - y2).abs()
    print("weight : ", pyconv.weight[0][0])
    print("bias : ", pyconv.bias[0])
    print("diff max : ", diff.max().item())
    print("diff mean: ", diff.mean().item())
    print("diff : ", (diff.max() / y1.abs().max()).item() * 100, "%")

```

raw result:

``` txt
[01/04/2024-04:24:40] [TRT] [W] BuilderFlag::kENABLE_TACTIC_HEURISTIC has been ignored in this builder run. This feature is only supported on Ampere and beyond.
[01/04/2024-04:24:42] [TRT] [W] Using default stream in enqueue()/enqueueV2()/enqueueV3() may lead to performance issues due to additional cudaDeviceSynchronize() calls by TensorRT to ensure correct synchronizations. Please use non-default stream instead.
weight :  tensor([[-0.0069,  0.0059,  0.0008],
        [ 0.0030, -0.0042,  0.0088],
        [-0.0020, -0.0030,  0.0008]], device='cuda:0', dtype=torch.float16,
       grad_fn=<SelectBackward0>)
bias :  tensor(-0.0025, device='cuda:0', dtype=torch.float16,
       grad_fn=<SelectBackward0>)
diff max :  0.025390625
diff mean:  0.0019474029541015625
diff :  0.958251953125 %
```

I'm confused with this accuracy difference. Pure calculation diff shouldn't be nearly 1% in just one conv layer.

Is this an issue, or just an usage error?

**Comment by zhaoxudong01** [comment]
the same issue, diff value will larger than 90% while the output_channel greater than 128.

**Comment by poweiw** [comment]
Hello @beegerous ! Is this issue still relevant?
